core-web/src/main/java/org/silverpeas/core/webapi/upload/ComponentInstanceFileUploadVerification.java

```diff
@@ -0,0 +1,101 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.silverpeas.core.webapi.upload;
+
+import org.silverpeas.core.admin.component.model.ComponentInstLight;
+import org.silverpeas.core.admin.service.OrganizationController;
+import org.silverpeas.core.io.upload.UploadedFile;
+import org.silverpeas.core.util.ServiceProvider;
+
+import java.io.File;
+import java.util.Optional;
+
+/**
+ * It is a process implied within the manufacturing of a new file upload.
+ * </p>
+ * When a file upload is performed, it is first verified according to its name (the file is not
+ * yet on the server), then is is verified a second time after than the upload is completed
+ * successfully. In some circumstances, according to the application, some verifications have to be
+ * performed in the behalf of the new file upload. The file upload process is unaware of these
+ * circumstances and it cannot know what verification to perform; It is the responsibility of the
+ * application to perform such verifications. This is why an implementation of this interface
+ * qualified by a name that satisfies the following convention <code>[COMPONENT
+ * NAME]InstanceFileUploadVerification</code> is looked for by the file upload process and then
+ * invoked if it is has been found.
+ * </p>
+ * Any application that requires specific actions when a file upload is performed has to implement
+ * this interface and the implementation has to be qualified with the @Named annotation by a name
+ * satisfying the following convention <code>[COMPONENT NAME]FileUploadVerification</code>. For
+ * example, for an application Kmelia, the implementation must be qualified with
+ * <code>@Named("kmeliaInstanceFileUploadVerification") </code>
+ * @author Yohann Chastagnier
+ */
+public interface ComponentInstanceFileUploadVerification {
+
+  /**
+   * The predefined suffix that must compound the name of each implementation of this interface.
+   * An implementation of this interface by a Silverpeas application named Kmelia must be named
+   * <code>kmelia[NAME_SUFFIX]</code> where NAME_SUFFIX is the predefined suffix as defined below.
+   */
+  String NAME_SUFFIX = "InstanceFileUploadVerification";
+
+  /**
+   * Gets the implementation of this interface with the qualified name guessed from the given
+   * component instance identifier.
+   * @param componentInstanceId the component instance identifier from which the qualified name of
+   * the implementation is guessed.
+   * @return either an implementation of this interface or nothing.
+   */
+  static Optional<ComponentInstanceFileUploadVerification> get(String componentInstanceId) {
+    try {
+      final ComponentInstLight componentInstLight =
+          OrganizationController.get().getComponentInstLight(componentInstanceId);
+      final String name = componentInstLight.getName() + NAME_SUFFIX;
+      return Optional.of(ServiceProvider.getService(name));
+    } catch (IllegalStateException ex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)